test/utils/clonableGenerator.js

```diff
@@ -0,0 +1,156 @@
+import test from 'tape';
+import { spy } from 'sinon';
+import { cloneableGenerator } from '../../src/internal/utils';
+
+test('it should proxy the original generator', assert => {
+  assert.plan(4);
+
+  const spyGenerator = {
```

Comment:
According to the MDN,

> The function* declaration (function keyword followed by an asterisk) defines a generator function, which returns a Generator object.

> The Generator object is returned by a generator function and it conforms to both the iterable protocol and the iterator protocol.

So it's kind of both :)
In my original code, it was called iterator but looking at MDN, I renamed it generator.
I you think its best calling it iterator, i will, i just wanted to put it outhere